auth/api/iam/jar.go

```diff
@@ -0,0 +1,172 @@
+package iam
+
+import (
+	"context"
+	"crypto"
+	"errors"
+	"net/url"
+
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/nuts-node/auth"
+	"github.com/nuts-foundation/nuts-node/auth/oauth"
+	cryptoNuts "github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/vdr/resolver"
+)
+
+// requestObjectModifier is a function that modifies the Claims/params of an unsigned or signed (JWT) OAuth2 request
+type requestObjectModifier func(claims map[string]string)
+
+type requestObject struct {
```

Comment:
why not have a SignedRequestObject (which is always signed)? Then; `jar.Sign(parameters) -> signedRequestObject`

(then less can go wrong when forgetting checks/using the incorrect state)